sw/device/sca/aes_serial/aes_serial.c

```diff
@@ -89,6 +89,116 @@ typedef enum simple_serial_result {
+  base_uart_stdout(&uart);
+}
+
+/**
+ * Initializes the GPIO peripheral.
+ */
+static void sca_init_gpio() {
+  dif_gpio_params_t gpio_params = {
+      .base_addr = mmio_region_from_addr(TOP_EARLGREY_GPIO_BASE_ADDR)};
+  (void)dif_gpio_init(gpio_params, &gpio);
+  (void)dif_gpio_output_set_enabled_all(&gpio, kGpioCaptureTriggerHigh);
+}
+
+/**
+ * Initializes the timer peripheral.
+ */
+static void sca_init_timer() {
+  (void)dif_rv_timer_init(
+      mmio_region_from_addr(TOP_EARLGREY_RV_TIMER_BASE_ADDR),
+      (dif_rv_timer_config_t){.hart_count = 1, .comparator_count = 1}, &timer);
+  dif_rv_timer_tick_params_t tick_params;
+  (void)dif_rv_timer_approximate_tick_params(kClockFreqPeripheralHz,
+                                             kClockFreqCpuHz, &tick_params);
+  (void)dif_rv_timer_set_tick_params(&timer, kRvTimerHart, tick_params);
+  (void)dif_rv_timer_irq_enable(&timer, kRvTimerHart, kRvTimerComparator,
+                                kDifRvTimerEnabled);
+  irq_timer_ctrl(true);
+  irq_global_ctrl(true);
+}
+
+/**
+ * Timer IRQ handler.
+ *
+ * Disables the counter and clears pending interrupts.
+ */
+void handler_irq_timer(void) {
+  // Return values of below functions are ignored to improve capture
+  // performance.
+  (void)dif_rv_timer_counter_set_enabled(&timer, kRvTimerHart,
+                                         kDifRvTimerDisabled);
+  (void)dif_rv_timer_irq_clear(&timer, kRvTimerHart, kRvTimerComparator);
+}
+
+/**
+ * Initializes the peripherals (pinmux, uart, gpio, and timer) used by SCA code.
+ */
+static void sca_init(void) {
+  pinmux_init();
+  sca_init_uart();
+  sca_init_gpio();
+  sca_init_timer();
+}
+
+/**
+ * Sets capture trigger high.
+ *
+ * The actual trigger signal used for capture is a combination (logical AND) of:
+ * - GPIO15 enabled here, and
+ * - the busy signal of the AES unit. This signal will go high 40 cycles
+ *   after aes_manual_trigger() is executed below and remain high until
+ *   the operation has finished.
+ */
+static void sca_set_trigger_high() {
+  (void)dif_gpio_write_all(&gpio, kGpioCaptureTriggerHigh);
+}
+
+/**
+ * Sets capture trigger low.
+ */
+static void sca_set_trigger_low() {
+  (void)dif_gpio_write_all(&gpio, kGpioCaptureTriggerLow);
+}
+
+/**
+ * Functions called by `sca_call_and_sleep()` must conform to this prototype.
+ */
+typedef void (*sca_callee)(void);
+
+static void sca_call_and_sleep(sca_callee callee, uint32_t sleep_cycles) {
+  // Start timer to wake Ibex after AES is done.
+  uint64_t current_time;
+  // Return values of below functions are ignored to improve capture
+  // performance.
+  (void)dif_rv_timer_counter_read(&timer, kRvTimerHart, &current_time);
+  (void)dif_rv_timer_arm(&timer, kRvTimerHart, kRvTimerComparator,
+                         current_time + sleep_cycles);
+  (void)dif_rv_timer_counter_set_enabled(&timer, kRvTimerHart,
+                                         kDifRvTimerEnabled);
+
+  callee();
```

Comment:
You may want to add a comment here saying that this is typically used to trigger the operation.

Reply:
Sure, but I think this comment belongs to the doxygen comment in the header file.